cmake/onnxruntime.cmake

```diff
@@ -122,7 +122,7 @@ else()
   else()
     onnxruntime_add_shared_library(onnxruntime ${CMAKE_CURRENT_BINARY_DIR}/generated_source.c )
   endif()
-  if(NOT APPLE)
+  if(NOT APPLE AND NOT ${CMAKE_SYSTEM_NAME} MATCHES "AIX")
```

Comment:
Can we use https://cmake.org/cmake/help/latest/module/CheckLinkerFlag.html instead? That would be more general.

Reply:
Or detect https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_COMPILER_LINKER_ID.html ?

Reply:
@snnn
made the suggested changes and using  check_linker_flag for checking linker support for path